packages/cms-lib/fileManager.js

```diff
@@ -186,52 +202,106 @@ async function fetchFolderContents(portalId, folder, dest, options) {
 }
 
 /**
- * Fetch a file/folder and write to local file system.
+ * Download a folder and write to local file system.
  *
  * @param {number} portalId
  * @param {string} src
  * @param {string} dest
+ * @param {object} file
  * @param {object} options
  */
-async function downloadFileOrFolder(portalId, src, dest, options) {
-  const { file, folder } = await getStat(portalId, src);
-  if (file) {
+async function downloadFolder(portalId, src, dest, folder, options) {
+  if (isHidden(folder)) {
+    logger.error('Folder "%s" could not be found in portal %s', src, portalId);
+    return;
```

Comment:
I thought about filtering out results at the request level (in `/api/filemanager`) but it seemed better to add messaging for the user similar to a regular 404.

Reply:
I think I agree when it comes to the top-level folder, but I think we should be clearer with the messaging so that we're differentiating hidden folders from not found.